pkg/cgu/cgu.go

```diff
@@ -26,6 +33,94 @@ type CguBuilder struct {
 	errorMsg string
 }
 
+// NewCguBuilder creates a new instance of CguBuilder.
+func NewCguBuilder(
+	apiClient *clients.Settings,
+	name, nsname string,
+	maxConcurrency int) *CguBuilder {
+	glog.V(100).Infof(
+		"Initializing new CGU structure with the following params: %s, %s",
+		name, nsname)
+
+	builder := CguBuilder{
+		apiClient: apiClient,
+		Definition: &v1alpha1.ClusterGroupUpgrade{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ClusterGroupUpgrade",
+				APIVersion: v1alpha1.SchemeGroupVersion.Version,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+			Spec: v1alpha1.ClusterGroupUpgradeSpec{
+				RemediationStrategy: &v1alpha1.RemediationStrategySpec{
+					MaxConcurrency: maxConcurrency,
+				},
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the CGU is empty")
+
+		builder.errorMsg = "CGU 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the CGU is empty")
+
+		builder.errorMsg = "CGU 'nsname' cannot be empty"
+	}
+
+	return &builder
+}
+
+// WithClusters sets clusters list in the CGU definition.
+func (builder *CguBuilder) WithClusters(clusterList []string) *CguBuilder {
```

Comment:
I think we need to validate that clusterList is not empty list.

Reply:
Would it be simpler if it is just WithCluster that accepts a single cluster string and appends it to builder.Definition.Spec.Clusters? Otherwise, you would have to check that the list is not empty and that each item in the list isn't an empty string to fully validate it.

Reply:
Yup, updated to use WithCluster instead.